src/cluster_legacy.c

```diff
@@ -5362,6 +5363,22 @@ void clusterCloseAllSlots(void) {
  * Cluster state evaluation function
  * -------------------------------------------------------------------------- */
 
+void clusterLogFailReason(int reason) {
+    if (reason == CLUSTER_FAIL_NONE) return;
+
+    char *msg;
+    switch (reason) {
+    case CLUSTER_FAIL_NOT_FULL_COVERAGE:
+        msg = "At least one hash slot is not served by any available node. "
+              "Please check the 'cluster-require-full-coverage' configuration.";
+        break;
+    case CLUSTER_FAIL_MINORITY_PARTITION: msg = "I am part of a minority partition."; break;
+    default: serverPanic("Unknown fail reason code.");
+    }
+    server.cluster->fail_reason = reason;
```

Comment:
maybe move "server.cluster->fail_reason = reason;" out of this function and in its caller. The name `clusterLogFailReason` suggests that this is a stateless logging function so it is a bit odd to have it update the server state. It does mean that we would need to "duplicate" this line but I personally feel that the code is more readable that way.

Reply:
yean, i can move it. Noted clusterLogCantFailover also do the same thing, it change the server state in the clusterLogCantFailover function itself.

Reply:
yeah - there was an apparently different bar back then :)